### PR TITLE
Fix #803: Add "follower_mode" to the config

### LINKED ISSUE #803: [CRDT] Untrusted Peers can remove pins and this causes weird situations
Untrusted peers can gain control of pins if they are the only ones allocated for that pin.
Forced example
```
$ ipfs-cluster-ctl -l <trusted_peer> pin add -r 1 --allocs <untrusted_peer> <hash>
$ ipfs-cluster-ctl -l <untrusted_peer> pin rm <hash>
# works!!!
$ ipfs-cluster-ctl -l <trusted_peer> pin ls
<hash> |  | PIN | Repl. Factor: 1--1 | Allocations: [<untrusted_peer>] | Recursive
# trusted peers still think it is around
```

This is an issue in the case of a cluster that has more untrusted peers than trusted peers, which is the model we are going for.
In a cluster that has a default replication of 3, uses the freespace allocator, has 10 peers, 4 trusted, 6 untrusted, once 3 of the untrusted peers have more freespace than the trusted peers they will get allocated all the pins and will have full control over those pins, with the trusted peers having no idea whether the pin is actually on the untrusted peer. 

//cc @hsanjuan we need to figure out how we are going to approach this scenario

### PR DESCRIPTION
**Pull request by hsanjuan**
Peers configured with follower_mode = true fail to add/pin/unpin.

Additionally they do not contact other peers when doing Status, Sync or
Recover and report on themselves.

They still contact other peers when doing "peers ls", as this is an OpenRPC
endpoint.

This is merely improving user interaction with a cluster peer and avoids
getting into confusing places:

* pin/unpin seems to work even no one trusts them
* status will query all peers in the peerset only to get auth errors and
ignore them, becoming way slower than it could be

This is not a security feature.

#803

### CODE REVIEW

**Comment by hsanjuan** [comment]
I should have left this as Draft PR. I put `FollowerMode` in the `cluster` config to use it avoid doing broadcasts and prevent peers from trying to add things, but this increases complexity vs putting it as part of the `crdt` component.

We could have an additional `BroadcastPeers` method in the crdt to use for broadcasts except `peers()` but that seems a bit weird too.

**Comment by kishansagathiya** [comment]
Is this ready for review?

**Comment by hsanjuan** [comment]
now yes

File: cluster.go [hunk 1]
```diff
@@ -43,6 +43,10 @@ const (
 	reBootstrapInterval = 30 * time.Second
 )
 
+var (
+	errFollowerMode = errors.New("this peer is configured as in follower mode. Write operations are disabled")
```
Comment (kishansagathiya) [inline cluster.go]:
```suggestion
	errFollowerMode = errors.New("this peer is configured to be in follower mode. Write operations are disabled")
```

File: ipfscluster_test.go [hunk 2]
```diff
@@ -1897,3 +1898,84 @@ func TestClustersDisabledRepinning(t *testing.T) {
...
+		}
+	})
+
+	t.Run("follower status itself only", func(t *testing.T) {
+		gpi, err := clusters[1].Status(ctx, test.Cid1)
+		if err != nil {
+			t.Error("status should work")
+		}
+		if len(gpi.PeerMap) != 1 {
+			t.Fatal("globalPinInfo[0] should only have one peer")
```
Comment (kishansagathiya) [inline ipfscluster_test.go]:
```suggestion
			t.Fatal("globalPinInfo should only have one peer")
```

File: ipfscluster_test.go [hunk 3]
```diff
@@ -1897,3 +1898,84 @@ func TestClustersDisabledRepinning(t *testing.T) {
...
+	defer shutdownClusters(t, clusters, mock)
+
+	_, err := clusters[0].Pin(ctx, test.Cid1, api.PinOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = clusters[0].Pin(ctx, test.ErrorCid, api.PinOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (kishansagathiya) [inline ipfscluster_test.go]: don't really need above two tests.

Comment (hsanjuan) [reply ipfscluster_test.go]: yes, the above is needed so the `sync` test below works.

**Pull request merged**